flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,456 @@
+                avroSchema.getType(), "The avro schema of the record must have a type.");
+        if (avroSchema.getType() != org.apache.avro.Schema.Type.RECORD) {
+            throw new IllegalArgumentException(
+                    "The Avro Schema must be of RECORD type to convert to BigQuery columns. Found Schema: "
+                            + avroSchema);
+        }
+        Preconditions.checkState(
+                !avroSchema.getFields().isEmpty(),
+                "The avro record schema must have at least one field.");
+        // Iterate over each record field and add them to the BigQuery schema.
+        List<Field> fields =
+                avroSchema.getFields().stream()
+                        .map(
+                                avroField -> {
+                                    Preconditions.checkNotNull(
+                                            avroField.name(),
+                                            "The avro field must have a name attribute.");
+                                    Field.Builder bigQueryFieldBuilder =
+                                            convertAvroFieldToBigQueryField(
+                                                            avroField.schema(), avroField.name(), 0)
+                                                    .toBuilder();
+                                    bigQueryFieldBuilder.setName(avroField.name());
+                                    if (avroField.doc() != null) {
+                                        bigQueryFieldBuilder.setDescription(avroField.doc());
+                                    }
+                                    return bigQueryFieldBuilder.build();
+                                })
+                        .collect(Collectors.toList());
+        return Schema.of(fields);
+    }
+
+    /**
+     * Converts an Avro field to a BigQuery field.
+     *
+     * @param avroField the Avro field to convert
+     * @param name the name of the field
+     * @param nestedLevel the current nesting level of the field
+     */
+    private static Field convertAvroFieldToBigQueryField(
+            org.apache.avro.Schema avroField, String name, int nestedLevel) {
+        if (nestedLevel > MAX_NESTED_LEVEL) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "BigQuery fields can only be nested 15 times. Found nesting level of: %s",
+                            nestedLevel));
+        }
+        Preconditions.checkNotNull(avroField, "The avro field cannot be null.");
+        Preconditions.checkNotNull(avroField.getType(), "The avro field type cannot be null.");
+        switch (avroField.getType()) {
+            case RECORD:
+                return convertAvroRecordFieldToBigQueryField(avroField, name, nestedLevel);
+            case ARRAY:
+                return convertAvroRepeatedFieldToBigQueryField(
+                        avroField.getElementType(), name, nestedLevel);
```

Comment:
For consistency, send the `avroField` as parameter instead of `element type`. The method can internally extract the element type as needed.

Reply:
Done